### PR TITLE
chore: remove nextui org

### PR DESCRIPTION
**Pull request by manika-signoz**
## Summary

Removes the `@nextui-org/react` dependency and the NextUI Tailwind plugin. Replaces NextUI usage with shared UI built on **Headless UI** (dialogs), **Radix UI** (popover, select, slider, tooltip), and small wrapper components (`AppModal`, `AppTooltip`, `PricingRangeSlider`). Wraps the app in Radix `TooltipProvider` for consistent tooltip behavior.

## Motivation / Problem

NextUI added a large dependency footprint and a separate theming/plugin path in Tailwind. Moving to lighter, composable primitives reduces bundle size, aligns styling with existing Tailwind tokens, and keeps modal/tooltip/slider behavior explicit and testable.

## Changes

- **Dependencies:** Remove `@nextui-org/react`; add `@radix-ui/react-popover`, `@radix-ui/react-slider`, `@radix-ui/react-tooltip` (alongside existing `@radix-ui/react-select`).
- **Tailwind:** Drop NextUI plugin and `node_modules/@nextui-org/theme` content paths; add `nav-popover-in` / `nav-popover-out` keyframes for nav dropdown motion.
- **App shell:** `ThemeProviders` wraps children with `TooltipProvider` (`delayDuration={400}`).
- **New/shared UI:** `components/ui/Modal.tsx` (`AppModal`), `AppTooltip.tsx`, `PricingRangeSlider.tsx`.
- **Migrated call sites:** Global nav (`TopNav`), `why-signoz` Book a Demo modal, pricing flows (calculator, monthly estimate, enterprise, metrics cost estimation), launch week, changelog header, docs sidebar, article series, video modals, tags with tooltips, datadog migration tool, merchandise section, home page wiring, etc.

## Description

This PR completes the NextUI removal by swapping each former NextUI component for Headless UI or Radix-based equivalents, with consistent modal close behavior (including backdrop and close control) and updated slider/tooltip UX where those primitives changed.

## Type of Change

- [ ] **Docs** – Changes to `data/docs/**`
- [ ] **Blog** – Changes to `data/blog/**`
- [x] **Site Code** – Changes to `app/**`, `components/**`, `hooks/**`, `utils/**`, config, etc.
- [ ] **Redirects** – Renamed/moved docs or updated `next.config.js` redirects
- [x] **Dependencies / CI / Scripts**
- [ ] **Other** – Explain below

## Impact

- [ ] Documentation only
- [x] UI changes
- [x] Developer workflow (dependency and Tailwind config changes)
- [x] Performance impact
- [ ] Breaking change

## Context & Screenshots
TopNav
Screenshot 2026-03-27 at 16 03 12
Screenshot 2026-03-27 at 16 03 33

Slider

https://github.com/user-attachments/assets/9fbfc3c6-5769-4cef-b095-676e0b8c27bf

https://github.com/user-attachments/assets/1b8e2dce-127b-479a-bb2a-796045b2d0b8

Why signoz modal
Screenshot 2026-03-27 at 16 07 20

Screenshot 2026-03-27 at 16 07 08

metrics-cost-estimation section 
https://github.com/user-attachments/assets/90461e6e-58a8-48aa-ba6a-bd88eba31dcb

Docs tags
https://signoz.io/docs/logs-management/overview/
Screenshot 2026-03-27 at 16 10 22

https://signoz-web-git-chore-nextui-org-react-signoz.vercel.app/docs/logs-management/overview/
Screenshot 2026-03-27 at 16 10 40

ArticleSeriesTop
Screenshot 2026-03-27 at 16 11 48

Screenshot 2026-03-27 at 16 11 55

enterprise page monthly estimate slider

https://github.com/user-attachments/assets/8a44c43a-8683-4f3c-9fcc-094e5c16213d

## Before / After (if applicable)

- **Before:** NextUI components + NextUI Tailwind plugin and theme paths.
- **After:** Headless UI + Radix primitives + shared `AppModal` / `AppTooltip` / `PricingRangeSlider`, standard Tailwind plugins only.

### General

- [x] Branch is up to date with `main`
- [x] PR title follows conventional format (`type: description`)
- [x] Self-reviewed the code
- [ ] Comments added for complex logic

### For all changes

- [ ] Built locally (`yarn build`) with no errors
- [ ] Ran `yarn lint` and fixed any issues
- [ ] Pre-commit hooks passed (or ran `yarn check:doc-redirects` / `yarn check:docs-metadata` if applicable)

### For site code changes

- [ ] Follows [Site Code Guidelines](CONTRIBUTING.md#website-code-guidelines) in CONTRIBUTING.md
- [x] New dependencies are justified in the PR description (Radix popover/slider/tooltip replace NextUI; aligns with existing `@radix-ui/react-select`)

### For renamed or moved docs

- [ ] N/A

## Testing

- [x] Tested locally
- [x] Edge cases considered (modal open/close, backdrop, tooltips, sliders)
- [x] Existing functionality verified
- [ ] New tests added (if applicable)

**Steps to test (manual):**
### 1. `AppModal` (`components/ui/Modal.tsx`)

| URL | What uses the modal |
|-----|---------------------|
| `/` | `VideoModalPlayer` (home hero video) |
| `/why-signoz` | `BookADemoModal` (Book a demo) |
| `/pricing/metrics-cost-estimation` | `MetricsCostEstimation` (e.g. video explainer modal) |
| `/pricing` | _(none — this page does not import `AppModal`)_ |
| `/launch-week` | `LaunchWeek` |
| `/datadog-migration-tool` | `DatadogMigrationTool` |
| `/events/kubecon-cloudnativecon-north-america-2025` | `MerchandiseSection` (merch / disclosure flow) |
| `/changelog` | `ChangelogHeader` (email subscribe modal) |
| `/changelog/[slug]` | `ChangelogHeader` (same modal; `showFilters={false}` on detail) |

---

### 2. `AppTooltip` (`components/ui/AppTooltip.tsx`)

**A. Docs shell — `/docs/**` (all documentation pages under the main docs app layout)**

- **`DocsSidebar`**: sidebar nav items on docs pages (`app/docs/(main-docs)/layout.tsx` → `DocsSidebar`).
- **`TagsWithTooltips`**: tag chips above the title on doc pages **when** `docTags` is non-empty (`components/DocContent/DocContent.tsx`). In Contentlayer, `docTags` is derived from frontmatter `tags` or falls back to defaults, so **most** doc pages show the tag row (spot-check any slug under `/docs/...`).

**Exact non-docs URLs**

| URL | Component |
|-----|-----------|
| `/pricing` | `TeamsPricingCard` (`app/pricing/pricingv1/components/TeamsPricingCard.tsx`) |
| `/enterprise` | `MonthlyEstimate` / `MonthlyEstimateMobile` inside `EnterprisePricing` (`components/EnterprisePricing/index.tsx`) |
| `/upgrade-path` | `DocContent` when a doc is loaded in the tool (`app/upgrade-path/components/DocRender.tsx`) — same tag tooltips as docs when tags apply |

**Blog posts using `ArticleSeriesTop` (tooltip on series link)**

| URL |
|-----|
| `/blog/opentelemetry-nextjs` |
| `/blog/opentelemetry-nextjs-use-cases` |
| `/blog/opentelemetry-nextjs-logging` |
| `/blog/opentelemetry-nextjs-web-vitals` |
| `/blog/opentelemetry-nextjs-production` |

**Guides using `ArticleSeriesTop`**

| URL |
|-----|
| `/guides/prometheus-node-exporter` |
| `/guides/prometheus-node-exporter-metrics` |
| `/guides/aws-elb-monitoring` |
| `/guides/aws-elb-metrics` |
| `/guides/aws-elb-logs` |
| `/guides/cloudwatch-pricing` |
| `/guides/cloudwatch-cost-optimization` |
| `/guides/vpc-flow-logs` |
| `/guides/vpc-flow-logs-aws` |

---

### 3. `PricingRangeSlider` (`components/ui/PricingRangeSlider.tsx`)

| URL | Where |
|-----|--------|
| `/pricing` | `PricingCalculator` (`app/pricing/pricingv1/components/PricingCalculator.tsx`) |
| `/enterprise` | `MonthlyEstimate` / `MonthlyEstimateMobile` (`components/Monthly-estimate/*`) via `EnterprisePricing` |
| `/pricing/metrics-cost-estimation` | `MetricsCostEstimation` |

---

**Note:** Submit as Draft by default. Mark "Ready for review" when checks pass and content is ready.

### CODE REVIEW

File: components/ui/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  backdrop = 'default',
+  showCloseButton = true,
+}: AppModalProps) {
+  return (
+    <Dialog
+      as="div"
+      className="bg-overlay/30 fixed inset-0 z-50 h-screen w-screen backdrop-blur-md backdrop-saturate-150"
+      open={isOpen}
+      onClose={() => onOpenChange(false)}
+      transition
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply components/ui/Modal.tsx:46]: fixed

File: components/ui/PricingRangeSlider.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  const [isDragging, setIsDragging] = useState(false)
+
+  const body = (
+    <>
+      <SliderPrimitive.Root
+        className="relative flex w-full touch-none select-none items-center py-2"
+        value={[value]}
+        onValueChange={(v) => onChange(v[0] ?? min)}
+        onPointerDown={() => setIsDragging(true)}
+        onPointerUp={() => setIsDragging(false)}
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply components/ui/PricingRangeSlider.tsx:88]: fixed

File: components/ui/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            )}
+            {children}
+          </DialogPanel>
+        </div>
+      </div>
+    </Dialog>
+  )
+}
+
+export function useDisclosure(defaultOpen = false) {
```
Comment (YounixM) [inline components/ui/Modal.tsx]: move to separate file.

File: app/pricing/metrics-cost-estimation/MetricsCostEstimation.tsx [hunk 4]
```diff
@@ -256,18 +191,20 @@ const MetricsCostEstimation = () => {
...
-                    </Link>
-                  </Button>
-                  <Button className="w-full" type={Button.TYPES.SECONDARY}>
-                    <Link href={'/docs/introduction/'} className="flex-center">
-                      Read the docs
-                      <ArrowRight size={14} />
-                    </Link>
-                  </Button>
+                  <Link
+                    href="/teams/"
```
Comment (YounixM) [inline app/pricing/metrics-cost-estimation/MetricsCostEstimation.tsx:196]: Create constants for common routes

File: components/ui/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+              'relative w-full transform overflow-hidden text-left align-middle shadow-xl transition duration-200 ease-out data-[closed]:scale-95 data-[closed]:opacity-0',
+              sizeClass[size],
+              panelClassName
+            )}
+          >
+            {showCloseButton && (
+              <button
+                type="button"
+                aria-label="Close modal"
+                className="absolute right-1 top-1 select-none appearance-none rounded-full p-2 text-zinc-400 outline-none transition-[background-color,color] [-webkit-tap-highlight-color:transparent] hover:bg-zinc-700/40 focus-visible:z-10 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-signoz_robin-500 active:bg-zinc-600/40"
```
Comment (YounixM) [inline components/ui/Modal.tsx:70]: Are these styles reusable ? We should consider creating components if styles are reused

File: package.json
Comment (YounixM) [inline package.json:22]: Evaluate the usage of headless ui library. Create a separate ticket for this if required.

File: package.json [hunk 6]
```diff
@@ -23,8 +23,10 @@
     "@heroicons/react": "^2.1.3",
```
Comment (YounixM) [inline package.json:23]: Do we need both hero icons and lucide icons library ?

**Review by YounixM — COMMENTED** [review]
Please check the review comments

File: components/ui/Modal.tsx (hunk 3 above)
Comment (manika-signoz) [reply components/ui/Modal.tsx]: done

File: app/pricing/metrics-cost-estimation/MetricsCostEstimation.tsx (hunk 4 above)
Comment (manika-signoz) [reply app/pricing/metrics-cost-estimation/MetricsCostEstimation.tsx:196]: we follow this pattern throughout the codebase, will pick this separately keeping pr scope in mind

File: components/ui/Modal.tsx (hunk 5 above)
Comment (manika-signoz) [reply components/ui/Modal.tsx:70]: only used in component for 'X' icon on modal, will extract to a component if reused

File: package.json
Comment (manika-signoz) [reply package.json:22]: will do, this was an existing dependency, will get back and evaluate

File: package.json (hunk 6 above)
Comment (manika-signoz) [reply package.json:23]: hmm, not needed, we can swap heroicons with lucide icons, thank you for pointing this out. let me pick this in a separate PR

File: app/pricing/metrics-cost-estimation/MetricsCostEstimation.tsx [hunk 7]
```diff
@@ -130,79 +126,19 @@ const MetricsCostEstimation = () => {
...
+                      <PricingRangeSlider
                         value={metricsValue}
                         onChange={handleChangeMetrics}
+                        min={MIN_VALUE}
+                        max={MAX_VALUE}
+                        step={1}
+                        color="secondary"
+                        minLabel="1"
+                        maxLabel="6"
+                        markLabels={['1', '2', '3', '4', '5', '6']}
```
Comment (YounixM) [inline app/pricing/metrics-cost-estimation/MetricsCostEstimation.tsx:138]: use constant

**Review by YounixM — APPROVED** [review]
Minor comment

File: app/pricing/metrics-cost-estimation/MetricsCostEstimation.tsx (hunk 7 above)
Comment (manika-signoz) [reply app/pricing/metrics-cost-estimation/MetricsCostEstimation.tsx:138]: noted, this and pricing calculator component have constants hardcoded, we should refactor this, will pick
